src/main/java/io/anserini/embeddings/search/IndexW2V.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * Anserini: An information retrieval toolkit built on Lucene
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.embeddings.search;
+
+import io.anserini.index.generator.LuceneDocumentGenerator;
+import io.anserini.rerank.ScoredDocuments;
+import io.anserini.util.AnalyzerUtils;
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.core.WhitespaceAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.FSDirectory;
+import org.kohsuke.args4j.*;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.FloatBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import static io.anserini.index.generator.LuceneDocumentGenerator.FIELD_BODY;
+import static io.anserini.index.generator.LuceneDocumentGenerator.FIELD_ID;
+
+public final class IndexW2V {
+  private static final Logger LOG = LogManager.getLogger(IndexW2V.class);
+
+  public static final class Args {
+
+    // required arguments
+    @Option(name = "-input", metaVar = "[Path]", required = true, usage = "collection path")
+    public String input;
+
+    @Option(name = "-index", metaVar = "[Path]", required = true, usage = "index path")
+    public String index;
+
+    @Option(name = "-term", metaVar = "String", usage = "get the embeddings corresponding to the term")
+    public String term = "";
+
+    //optional arguments
+    @Option(name = "--build", usage = "builds index if specified")
```

Comment:
This code is both builds and searches the index. An index is built if `--build` is specified. Double dash because the option doesn't take arguments.